php/EE/Migration/Executor.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace EE\Migration;
+
+use \EE;
+use \EE\Utils;
+use Symfony\Component\Finder\Finder;
+
+class Executor {
+
+    const MIGRATION_PATH = '/Users/kirtan/code/easyengine/easyengine/migrations/';
+
+    /**
+     * Executes all pending migrations
+     */
+    public static function execute_migrations() {
+
+        Utils\delem_log( "ee migration start" );
+        EE::log( "Migrating EasyEngine data to new version" );
+
+        $migrations = self::get_migrations_to_execute();
+
+        if( empty( $migrations ) ) {
+            EE::success( "Noting to migrate" );
+            exit( 0 );
+        }
+
+        sort( $migrations );
+
+        try {
+            self::execute_migration_stack( $migrations );
+        } catch( \Throwable $e ) {
+            Utils\delem_log( "ee migration ended abruptly" );
+            exit( 1 );
+        }
+
+        EE::success( "Successfully migrated EasyEngine" );
+    }
+
+    /**
+     * Executes all migrations passed to it recursively.
+     * Also undo'es all migration if there was error executing any migration
+     */
+    private static function execute_migration_stack( $migrations ) {
+        if( empty( $migrations ) ) {
+            return;
+        }
+        
+        $migration_path = self::get_migration_path( $migrations[0] );
+        $migration_class_name = self::get_migration_class_name( $migrations[0] );
+        
+        if( ! file_exists( $migration_path ) ) {
+            EE::error( "Unable to find migration file at $migration_path", false );
+            throw new Exception();
+        }
+
+        require( $migration_path );
+
+        try {
+            $migration = new $migration_class_name;
+            if( ! $migration instanceof Base ) {
```

Comment:
I think this is supposed to be check against $migration_class_name.

`if( ! $migration instanceof $migration_class_name ) {` 

You might not have faced issue because this classes have used Base abstract class but here we don't want to check that.

Reply:
yes, you're right we don't need to check instance here. I've misunderstood the functionlity here. 

I think I just meant to check if this `$migration_class_name` class is exists or not then create object from that class.
before L60 just check `if ( class_exists( $migration_class_name ) )`: this is when we use file name properly but didn't use class name properly. Here we have this in try catch so of course it will throw the error when the class is undefined. But we might able to handle/show this error with some custom message. ( I believe we should use try/catch to handle unknown errors, especially when we are using `Exeption` or `Throwable` in catch and not using some custom exeptions )

Reply:
Note: This is no blocker can be merged.